src/ITILTemplateReadonlyField.php

```diff
@@ -0,0 +1,265 @@
+     * @param $withtypeandcategory   boolean  with type and category (false by default)
+     *
+     * @return array of Readonly fields
+     **/
+    public function getReadonlyFields($ID, $withtypeandcategory = false)
+    {
+        global $DB;
+
+        $iterator = $DB->request([
+            'FROM'   => $this->getTable(),
+            'WHERE'  => [static::$items_id => $ID],
+            'ORDER'  => 'id'
+        ]);
+
+        $tt_class       = static::$itemtype;
+        $tt             = new $tt_class();
+        $allowed_fields = $tt->getAllowedFields($withtypeandcategory);
+        $fields         = [];
+
+        foreach ($iterator as $rule) {
+            if (isset($allowed_fields[$rule['num']])) {
+                $fields[$allowed_fields[$rule['num']]] = $rule['num'];
+            }
+        }
+        return $fields;
+    }
+
+
+    /**
+     * Return fields who doesn't need to be used for this part of template
+     *
+     * @return array the excluded fields (keys and values are equals)
+     */
+    public static function getExcludedFields()
+    {
+        return [
+            175 => 175, // ticket's tasks (template)
+            1   => 1,   // Title
+            21  => 21,  // Description
+            // Special cases, not yet handled:
+            4   => 4,   // Requester
+            5   => 5,   // Technician
+            6   => 6,   // Assigned to a supplier
+            8   => 8,   // Technician group
+            13  => 13,  // Associated elements
+            52  => 52,  // Approval
+            65  => 65,  // Observer group
+            66  => 66,  // Observer
+            71  => 71,  // Requester group
+            142 => 142, // Documents
+            193 => 193, // Contract
+        ];
+    }
+
+
+    /**
+     * Print the readonly fields
+     *
+     * @param ITILTemplate $tt            ITIL Template
+     * @param boolean      $withtemplate  Template or basic item (default 0)
+     *
+     * @return void
+     **/
+    public static function showForITILTemplate(ITILTemplate $tt, $withtemplate = 0)
+    {
+        global $DB;
+
+        $ID = $tt->fields['id'];
+
+        if (!$tt->getFromDB($ID) || !$tt->can($ID, READ)) {
+            return false;
+        }
+        $canedit = $tt->canEdit($ID);
+        $ttm     = new static();
+        $fields  = $tt->getAllowedFieldsNames(false);
+        $fields  = array_diff_key($fields, self::getExcludedFields());
+        $rand    = mt_rand();
+
+        $iterator = $DB->request([
+            'FROM'   => static::getTable(),
+            'WHERE'  => [static::$items_id => $ID]
+        ]);
+
+        $numrows = count($iterator);
+
+        $readonlyfields = [];
+        $used         = [];
+        foreach ($iterator as $data) {
+            $readonlyfields[$data['id']] = $data;
+            $used[$data['num']]        = $data['num'];
+        }
+
+        if ($canedit) {
+            echo "<div class='firstbloc'>";
```

Comment:
We should use twig templates for new developments; unless there is a reason not to.